frontend/store/auth.ts

```diff
@@ -0,0 +1,297 @@
+  currentServerIndex: number;
+  currentUserIndex: number;
+  users: UserDto[];
+  rememberMe: boolean;
+  /**
+   * Key: userId. Value: Access token
+   */
+  accessTokens: { [key: string]: string };
+}
+
+export const authStore = defineStore('auth', {
+  state: () => {
+    return {
+      servers: [],
+      currentServerIndex: -1,
+      currentUserIndex: -1,
+      users: [],
+      rememberMe: true,
+      accessTokens: {}
+    } as AuthState;
+  },
+  actions: {
+    /**
+     * Adds a new server to the store and sets it as the default one
+     *
+     * @param serverUrl
+     * @param isDefault
+     */
+    async connectServer(serverUrl: string, isDefault = false) {
+      serverUrl = serverUrl.replace(/\/$/, '');
+
+      const snackbar = snackbarStore();
+
+      this.$nuxt.$axios.defaults.baseURL = serverUrl;
+      this.setAxiosHeader();
+
+      let data: ServerInfo;
+
+      try {
+        data = (await this.$nuxt.$api.system.getPublicSystemInfo())
+          .data as ServerInfo;
+        data.PublicAddress = serverUrl;
+        data.isDefault = isDefault;
+      } catch (err) {
+        snackbar.push(this.$nuxt.i18n.t('login.serverNotFound'), 'error');
+        throw new Error(err as string);
+      }
+
+      const semverMajor = parseInt(data.Version?.split('.')[0] as string);
+      const semverMinor = parseInt(data.Version?.split('.')[1] as string);
+      const isServerVersionSupported =
+        semverMajor > 10 || (semverMajor === 10 && semverMinor >= 7);
+
+      if (isServerVersionSupported) {
+        if (!data.StartupWizardCompleted) {
+          this.$nuxt.redirect('/wizard');
+        } else {
+          if (!this.getServerById(data.Id)) {
+            this.servers.push(data);
+          }
+
+          this.currentServerIndex = this.servers.indexOf(
+            this.servers.find(
+              (serv) => serv.PublicAddress === serverUrl
+            ) as ServerInfo
+          );
+        }
+      } else {
+        snackbar.push(this.$nuxt.i18n.t('login.serverVersionTooLow'), 'error');
+        throw new Error('Server version is too low');
+      }
+    },
+    /**
+     * Logs the user to the current server
+     *
+     * @param username
+     * @param password
+     * @param rememberMe
+     */
+    async loginUser(username: string, password: string, rememberMe: boolean) {
+      if (!this.currentServer) {
+        throw new Error('There is no server in use');
+      }
+
+      try {
+        const authenticateResponse = (
+          await this.$nuxt.$api.user.authenticateUserByName({
+            authenticateUserByName: {
+              Username: username,
+              Pw: password
+            }
+          })
+        ).data;
+
+        this.rememberMe = rememberMe;
+
+        if (authenticateResponse.User?.Id && authenticateResponse.AccessToken) {
+          Vue.set(
+            this.accessTokens,
+            authenticateResponse.User.Id,
+            authenticateResponse.AccessToken
+          );
```

Comment:
What does it do?

Reply:
[This is a caveat in Vue 2 reactivity system](https://v2.vuejs.org/v2/guide/reactivity.html#For-Objects)

Basically, if you have an object like this:

```js
mainKey: {
    key1: 'string',
    key2: 3,
    key3: ['hello', 1]
}
```

Vue won't react to changes if you directly add the key:

```js
mainKey.key4 = 6;

## Which transforms the object to:
mainKey: {
    key1: 'string',
    key2: 3,
    key3: ['hello', 1],
    key4: 6
}
``` 
Or delete it:

```js
delete mainKey.key3;

## Which transforms the object to:
mainKey: {
    key1: 'string',
    key2: 3
}
```

This is why ``Vue.set`` and ``Vue.delete`` helper functions exist, so Vue can correctly proxy the "transformed" objects as observables.

Vue 3 with composition API has some improvements in this regards iirc, but I can't explain them in detail right now.